src/events/EventsSection.tsx

```diff
@@ -122,60 +230,269 @@ export const EventsSection = () => {
-                },
-                {
-                  name: "clientId",
-                  displayKey: "events:client",
-                },
-              ]}
-              emptyState={
-                <ListEmptyState
-                  message={t("emptyEvents")}
-                  instructions={t("emptyEventsInstructions")}
-                />
-              }
-            />
+            <Flex>
+              <FlexItem>
+                <Dropdown
+                  id="user-events-search-select"
+                  data-testid="UserEventsSearchSelector"
+                  toggle={
+                    <DropdownToggle
+                      data-testid="userEventsSearchSelectorToggle"
+                      onToggle={onDropdownToggle}
+                      className="keycloak__user_events_search_selector_dropdown__toggle pf-u-mt-md pf-u-ml-md pf-u-mb-md"
+                    >
+                      {t("searchForEvent")}
+                    </DropdownToggle>
+                  }
+                  isOpen={searchDropdownOpen}
+                >
+                  <Form
+                    isHorizontal
+                    className="keycloak__user_events_search__form"
+                    data-testid="searchForm"
+                  >
+                    <FormGroup
+                      label={t("userId")}
+                      fieldId="kc-userId"
+                      className="keycloak__user_events_search__form_label"
+                    >
+                      <TextInput
+                        ref={register()}
+                        type="text"
+                        id="kc-userId"
+                        name="userId"
+                        data-testid="userId-searchField"
+                        onChange={(_val, event) => {
+                          const { value } = event.currentTarget;
+                          setValue("userId", value);
+                          setIsDirty(true);
+                        }}
+                        defaultValue={
+                          formValues?.userId ? formValues?.userId : ""
+                        }
+                      />
+                    </FormGroup>
+                    <FormGroup
+                      label={t("eventType")}
+                      fieldId="kc-eventType"
+                      className="keycloak__user_events_search__form_label"
+                    >
+                      <Select
+                        id="kc-eventType"
+                        name="eventType"
+                        data-testid="event-type-searchField"
+                        chipGroupProps={{
+                          numChips: 1,
+                          expandedText: "Hide",
+                          collapsedText: "Show ${remaining}",
+                        }}
+                        variant={SelectVariant.typeaheadMulti}
+                        typeAheadAriaLabel="Select"
+                        onToggle={onSelectToggle}
```

Comment:
We don't need to manage the toggle state ourselves as the callback provides an argument for it.
```suggestion
                        onToggle={(isOpen) => setSelectOpen(isOpen)}
```